doc/mgr/orchestrator.rst

```diff
@@ -273,13 +273,85 @@ Placement Specification
 
 In order to allow the orchestrator to deploy a *service*, it needs to
 know how many and where it should deploy *daemons*. The orchestrator 
-defines a placement specification:
+defines a placement specification that can either be passed as command line arguments.
 
-* To deploy three *daemons*, simply specify the count: ``3``
-* To deploy *daemons* on specific hosts, specify the host names: ``host1 host2 host3``
-* To deploy *daemons* on a subset of hosts, also specify the count: ``2 host1 host2 host3``
+Explicit placements
+^^^^^^^^^^^^^^^^^^^
 
+Daemons can be explictly placed on hosts by simply specifying them::
+
+    orch apply prometheus "host1 host2 host3"
+    
+or in yaml::
+  
+    service_type: prometheus
+    placement:
+      hosts: "host1 host2 host3"
```

Comment:
I wonder if we want to support both syntactical versions of `hosts` definitions in the future.

```yaml
    service_type: prometheus
    placement:
      hosts: "host1 host2 host3"
```

vs

```yaml
    service_type: prometheus
    placement:
      hosts: 
        - host1
        - host2
        - host3
```

Reply:
> I wonder if we want to support both syntactical versions of `hosts` definitions in the future.
> 
> ```yaml
>     service_type: prometheus
>     placement:
>       hosts: "host1 host2 host3"
> ```

Indeed this doesn't really make sense. Either we parse the whole PlacemetSpec or we don't. 

```yaml
service_type: prometheus
placement: "3 label:monitoring"
```

or 

```yaml
service_type: prometheus
placement: 
  count: 3
  label: "monitoring"
  hosts:
    - host1
    - "host2:2001:0db8:85a3:0000:0000:8a2e:0370:733=yougettheidea"
```

I'm ok with both!

Reply:
or we recommend a very explicit form in the first place:

this is how we internally save the placement struct if a `hosts` struct was given:

``` yaml
placement:
  hosts:
  - hostname: cephadm-dev
    name: ''
    network: ''
```

A bit more verbose, but it's very descriptive. I can imagine people struggling with the syntax we used on the commandline (host:network=optional_name).

Reply:
imo too verbose :smile: